services/provider/server/server.go

```diff
@@ -681,12 +707,31 @@ func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.S
 				"clusterID":          getSubVolumeGroupClusterID(subVolumeGroup),
 				"subvolumegroupname": subVolumeGroup.Name,
 				"fsName":             subVolumeGroup.Spec.FilesystemName,
-				"pool":               subVolumeGroup.GetLabels()[v1alpha1.CephFileSystemDataPoolLabel],
+				"pool":               subVolumeGroup.GetLabels()[ocsv1alpha1.CephFileSystemDataPoolLabel],
 				"csi.storage.k8s.io/provisioner-secret-name":       provisionerSecretName,
 				"csi.storage.k8s.io/node-stage-secret-name":        nodeSecretName,
 				"csi.storage.k8s.io/controller-expand-secret-name": provisionerSecretName,
 			}
 
+			clientProfileSpec := &csiopv1a1.ClientProfileSpec{
+				CephFs: &csiopv1a1.CephFsConfigSpec{
+					SubVolumeGroup: cephRes.Name,
+				},
+			}
+			storageClusters := &ocsv1.StorageClusterList{}
+			if err := s.client.List(ctx, storageClusters, client.InNamespace(s.namespace), client.Limit(1)); err != nil {
+				return nil, status.Errorf(codes.Internal, "failed to get storage cluster: %v", err)
+			}
+			if len(storageClusters.Items) < 1 {
+				return nil, status.Errorf(codes.Internal, "storagecluster does not exist")
+			}
+			kernelMountOptions := util.GetCephFSKernelMountOptions(&storageClusters.Items[0])
```

Comment:
How do you know that storage cluster 0 is the correct one to take the configuration from?

Reply:
In our namespace we are only supporting 1 storagecluster.

Reply:
So why do you need to list?

Reply:
done.